manager/src/main/java/org/dromara/hertzbeat/manager/component/alerter/common/ContentRender.java

```diff
@@ -0,0 +1,143 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.dromara.hertzbeat.manager.component.alerter.common;
+
+import freemarker.cache.StringTemplateLoader;
+import freemarker.template.Configuration;
+import freemarker.template.TemplateException;
+import lombok.extern.slf4j.Slf4j;
+import org.dromara.hertzbeat.common.constants.CommonConstants;
+import org.dromara.hertzbeat.common.entity.alerter.Alert;
+import org.dromara.hertzbeat.common.entity.manager.NoticeTemplate;
+import org.dromara.hertzbeat.common.util.ResourceBundleUtil;
+import org.dromara.hertzbeat.manager.service.NoticeConfigService;
+import org.springframework.stereotype.Component;
+import org.springframework.ui.freemarker.FreeMarkerTemplateUtils;
+
+import javax.annotation.Resource;
+import java.io.IOException;
+import java.time.Instant;
+import java.time.ZoneId;
+import java.time.format.DateTimeFormatter;
+import java.util.HashMap;
+import java.util.Locale;
+import java.util.Map;
+import java.util.ResourceBundle;
+
+/**
+ * Created by liven on 2024/1/11
+ *
+ * @author liven
+ * @date 2024/1/11
+ */
+
+@Slf4j
+@Component
+public class ContentRender {
+
+    private static final String NUMBER_FORMAT = "0";
+
+    private static final String TEMPLATE_NAME = "freeMakerTemplate";
+
+    protected static final DateTimeFormatter DTF = DateTimeFormatter.ofPattern("yyyy-MM-dd HH:mm:ss");
+    private static ContentRender instance;
+
+    private Configuration cfg;
+
+    private StringTemplateLoader stringLoader;
+
+    private ResourceBundle bundle;
+
+    @Resource
+    protected NoticeConfigService noticeConfigService;
+
+    private ContentRender() {
+        stringLoader = new StringTemplateLoader();
+        cfg = new Configuration(Configuration.VERSION_2_3_23);
+        cfg.setNumberFormat(NUMBER_FORMAT);
+        bundle = ResourceBundleUtil.getBundle("alerter");
+    }
+
+    public static synchronized ContentRender getInstance() {
+        if (instance == null){
+            instance = new ContentRender();
+        }
+        return instance;
+    }
```

Comment:
👍👍Hi, It look good. There is a question, due the @Component, spring will create an `ContentRender ` instance, and there we new an ContentRender(), what the different, can we use the spring instance only?